src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -103,27 +104,34 @@ final class AddProductToOrderHandler extends AbstractOrderHandler implements Add
      * @var OrderDetailUpdater
      */
     private $orderDetailUpdater;
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
 
     /**
      * @param TranslatorInterface $translator
      * @param ContextStateManager $contextStateManager
      * @param OrderAmountUpdater $orderAmountUpdater
      * @param OrderProductQuantityUpdater $orderProductQuantityUpdater
      * @param OrderDetailUpdater $orderDetailUpdater
+     * @param ConfigurationInterface $configuration
      */
     public function __construct(
         TranslatorInterface $translator,
         ContextStateManager $contextStateManager,
         OrderAmountUpdater $orderAmountUpdater,
         OrderProductQuantityUpdater $orderProductQuantityUpdater,
-        OrderDetailUpdater $orderDetailUpdater
+        OrderDetailUpdater $orderDetailUpdater,
+        ConfigurationInterface $configuration
```

Comment:
If I am not wrong, this is a BC break because you add a new constructor argument without providing a default value.

Can we either
- mark it in PR and acknowledge it
- or use a default value (can be `null` and we return hardcoded `$shouldQuantityDiscountsBeApplied = 1`) ?

Reply:
That being said I think I remember something about handler constructors said recently (and BC breaks) 🤔

Reply:
Yep, my opinion was: OK to use the interface, which is better than the legacy class, but then let's accept that it's a BC break and use everywhere Because handling both use cases (with the service injected or without) only creates confusion and overly complicated code for no value

And changing the dependencies of a handler is acceptable even in a patch version, or else we could have never fixed any bug for the Order page, where we frequently need to inject the `OrderAmountUpdater` or the `ContextStateManager` and so

Adapter classes are not meant to be instanciated manually anyway, since they are used as middleware for the CQRS commands

TLDR: this is an acceptable BC break, we just need to indicate it in the PR description ^^